status/api/accounts.nim

```diff
@@ -90,13 +114,18 @@ proc storeDerivedAccounts(self: StatusObject, id: UUID, keyUid: string,
       self.closeUserDb()
 
     return PublicAccountResult.ok(pubAccount)
-  except Exception as e:
-    return PublicAccountResult.err e.msg
+  except AccountDbError as e:
+    return PublicAccountResult.err errorMsg & e.msg
+  except StatusApiError as e:
+    return PublicAccountResult.err errorMsg & e.msg
+  except ValueError as e:
+    return PublicAccountResult.err "Error parsing address of default wallet " &
+      "account: " & e.msg
 
 
 proc createAccount*(self: StatusObject, mnemonicPhraseLength: int,
   bip39Passphrase, password: string, dir: string): PublicAccountResult {.raises:
-  [ref Exception, ref ValueError].} =
+  [Defect, StatusApiError].} =
```

Comment:
another small detail worth considering: exceptions and `Result` are both error handling mechanisms that by and large address the same problem - we tend to avoid using both in the same function, else the calling code must go through some gymnastics to handle both types - one case were this potentially doesn't apply is if there's some truly exceptional construct that _should_ ignore all "normal" error handling and jump straight to an exception handler, but that's .. rare.

Reply:
Very good point and I did strive for that. Looks like this one slipped through. Will fix so that the StatusApiError is caught.

Reply:
Making sure I understand... so if the bases are covered inside the proc, then we don't need/want the raises pragma here (keeping in mind that at the top of the module we have `{.push raises: [Defect].}`)?

Reply:
> we don't need/want the raises pragma here

correct

Reply:
I believe I did catch StatusApiError in other top-level, public-facing procs. Really think I just missed this one. May be others as well, will review.